internal/graph/track_check_resolver_test.go

```diff
@@ -0,0 +1,154 @@
+	t.Cleanup(localChecker.Close)
+
+	trackChecker := NewTrackCheckResolver(
+		WithTrackerContext(ctx),
+		WithTrackerLogger(logger))
+	t.Cleanup(trackChecker.Close)
+
+	cycleDetectionCheckResolver.SetDelegate(trackChecker)
+	trackChecker.SetDelegate(localChecker)
+	localChecker.SetDelegate(cycleDetectionCheckResolver)
+
+	t.Run("tracker_delegates_request", func(t *testing.T) {
+		ds := memory.New()
+		t.Cleanup(ds.Close)
+
+		storeID := ulid.Make().String()
+
+		model := parser.MustTransformDSLToProto(`
+		model
+		  schema 1.1
+
+		type user
+
+		type group
+		  relations
+			define blocked: [user, group#member]
+			define member: [user, group#member] but not blocked
+`)
+
+		err := ds.Write(
+			ctx,
+			storeID,
+			nil,
+			[]*openfgav1.TupleKey{
+				tuple.NewTupleKey("group:1", "member", "user:jon"),
+				tuple.NewTupleKey("group:2", "blocked", "group:1#member"),
+				tuple.NewTupleKey("group:3", "blocked", "group:1#member"),
+			})
+		require.NoError(t, err)
+
+		typesys, err := typesystem.NewAndValidate(
+			context.Background(),
+			model,
+		)
+		require.NoError(t, err)
+
+		ctx = storage.ContextWithRelationshipTupleReader(ctx, ds)
+		ctx = typesystem.ContextWithTypesystem(ctx, typesys)
+		resp, err := trackChecker.ResolveCheck(ctx, &ResolveCheckRequest{
+			AuthorizationModelID: ulid.Make().String(),
+			StoreID:              storeID,
+			TupleKey:             tuple.NewTupleKey("group:1", "blocked", "user:jon"),
+			RequestMetadata:      NewCheckRequestMetadata(25),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.False(t, resp.GetAllowed())
+	})
+
+	t.Run("tracker_delegates_request", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		t.Cleanup(ctrl.Finish)
+
+		mockLocalChecker := NewMockCheckResolver(ctrl)
+		mockLocalChecker.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).Return(&ResolveCheckResponse{
+			Allowed: true,
+		}, nil).Times(1)
+		trackChecker.SetDelegate(mockLocalChecker)
+
+		resp, err := trackChecker.ResolveCheck(ctx, &ResolveCheckRequest{
+			StoreID:         ulid.Make().String(),
+			TupleKey:        tuple.NewTupleKey("document:1", "viewer", "user:will"),
+			RequestMetadata: NewCheckRequestMetadata(defaultResolveNodeLimit),
+			VisitedPaths:    map[string]struct{}{},
+		})
+
+		require.NoError(t, err)
+		require.True(t, resp.GetAllowed())
+	})
+
+	t.Run("tracker_user_type", func(t *testing.T) {
+		userType := trackChecker.userType("group:1#member")
+		require.Equal(t, "userset", userType)
+
+		userType = trackChecker.userType("user:ann")
+		require.Equal(t, "user", userType)
+
+		userType = trackChecker.userType("user:*")
+		require.Equal(t, "userset", userType)
+	})
+
+	t.Run("traacker_expires_paths", func(t *testing.T) {
+		r := resolutionTree{tm: time.Now().Add(-trackerInterval)}
+		require.True(t, r.expired())
+
+		r = resolutionTree{tm: time.Now().Add(trackerInterval)}
+		require.False(t, r.expired())
+	})
+
+	t.Run("tracker_prints_and_delete_path", func(t *testing.T) {
```

Comment:
I don't understand at all what this test is doing 😭 Also you are not making any assertions on `trackChecker`

Reply:
> I don't understand at all what this test is doing 😭 Also you are not making any assertions on `trackChecker`
Resolver only reports expired paths, this test verifies it.